x-pack/plugins/lens/public/datasources/form_based/datapanel.tsx

```diff
@@ -297,6 +300,14 @@ export const InnerFormBasedDataPanel = function InnerFormBasedDataPanel({
     data.search.session,
   ]);
 
+  useEffect(() => {
+    if (hasNewFields) {
+      refreshFieldList();
```

Comment:
Is it necessary to do in Lens if we have a refresh in `packages/kbn-unified-field-list/src/services/field_existing/field_existing_utils.ts`?

Reply:
yes, else lens isn't aware that there are new fields, therefore the field list refresh has to be triggered, else new fields don't show up